src/compiler/transformers/esDecorators.ts

```diff
@@ -726,8 +727,11 @@ export function transformESDecorators(context: TransformationContext): (x: Sourc
         // - The second pass visits the constructor to add instance initializers.
         //
         // NOTE: If there are no constructors, but there are instance initializers, a synthetic constructor is added.
-
         enterClass(classInfo);
+
+        classInfo.metadataReference = factory.createIdentifier("metadata");

```

Comment:
This could potentially clobber an outer variable named `metadata` that could be referred to by class element decorators. Use this instead:

```suggestion
        classInfo.metadataReference = factory.createUniqueName("_metadata", GeneratedIdentifierFlags.Optimistic | GeneratedIdentifierFlags.FileLevel);
```

I'd also recommend using a `_` prefix for consistency with other decorateor-related variables (it also reduces the likelihood of needing a numbered suffix.)

You should also ensure there is a test that shows this collision case. Something like:

```ts
// @target: es2022
declare var metadata: any;
class C {
  @metadata m() {}
}
```